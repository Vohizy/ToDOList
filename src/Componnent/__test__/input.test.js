import { fireEvent, render, screen } from "@testing-library/react"
import  renderer  from "react-test-renderer"
import Input from "../input"

describe('test for input',()=>{
    it("input there are a placeholder",() => {
        render(<Input />)
        const getInput = screen.getByTestId("input")
        expect(getInput).toHaveAttribute('placeholder', 'enter your text')
        
    })
    it('i put a word and input is react',() =>{
        render(<Input />)
        const getInput = screen.getByTestId("input")
        fireEvent.change(getInput,{target:{value:"qwert"}})
        const makeValueInInput = getInput.value
        expect(makeValueInInput).toBe("qwert")
    })
    it('prove when i press key Enter in input and a function reload',() => {
        const mock = jest.fn
        render(<Input KeyPress={mock}/>)
        const getInput = screen.getByTestId("input")
        fireEvent.keyPress(document ,{key : 'Enter'})
        expect(mock.call.length).toBe(1)
    })
    it('test of snapshot',() => {
        const mock = jest.fn
        const tree =  renderer.create(<Input Change={mock} KeyPress={mock} Input={mock}/>).toJSON() 
        expect(tree).toMatchSnapshot();
    })

})