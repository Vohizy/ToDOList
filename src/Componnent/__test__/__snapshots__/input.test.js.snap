// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test for input test of snapshot 1`] = `
<input
  className="form-control"
  data-testid="input"
  onChange={[Function]}
  onKeyPress={[Function]}
  placeholder="enter your text"
  type="text"
  value={[Function]}
/>
`;
